rma/models/rma.py

```diff
@@ -0,0 +1,772 @@
+        [('customer', 'Customer'), ('supplier', 'Supplier')],
+        string="Type", required=True, default=_get_default_type, readonly=True)
+    reference = fields.Char(string='Reference',
+                            help="The partner reference of this RMA order.",
+                            readonly=True,
+                            states={'draft': [('readonly', False)]})
+
+    comment = fields.Text('Additional Information', readonly=True, states={
+        'draft': [('readonly', False)]})
+    delivery_address_id = fields.Many2one(
+        'res.partner', readonly=True,
+        states={'draft': [('readonly', False)]},
+        string='Partner delivery address',
+        help="This address will be used to "
+        "deliver repaired or replacement products.")
+    invoice_address_id = fields.Many2one(
+        'res.partner', readonly=True,
+        required=True,
+        states={'draft': [('readonly', False)]},
+        string='Partner invoice address',
+        help="Invoice address for current rma order.")
+    add_invoice_id = fields.Many2one('account.invoice', string='Add Invoice',
+                                     ondelete='set null', readonly=True,
+                                     states={'draft': [('readonly', False)]})
+    state = fields.Selection([('draft', 'Draft'), ('to_approve', 'To Approve'),
+                              ('approved', 'Approved'),
+                              ('done', 'Done')], string='State', index=True,
+                             default='draft')
+    date_rma = fields.Datetime(string='RMA Date', index=True, copy=False)
+    partner_id = fields.Many2one('res.partner', string='Partner',
+                                 required=True, readonly=True,
+                                 states={'draft': [('readonly', False)]})
+    assigned_to = fields.Many2one('res.users', 'Assigned to',
+                                  track_visibility='onchange')
+    requested_by = fields.Many2one('res.users', 'Requested by',
+                                   track_visibility='onchange',
+                                   default=lambda self: self.env.user)
+    company_id = fields.Many2one('res.company', string='Company',
+                                 default=lambda self: self.env.user.company_id)
+    rma_line_ids = fields.One2many('rma.order.line', 'rma_id',
+                                   string='RMA lines',
+                                   copy=False)
+    warehouse_id = fields.Many2one('stock.warehouse', string='Warehouse',
+                                   required=True, readonly=True,
+                                   states={'draft': [('readonly', False)]},
+                                   default=_default_warehouse_id)
+
+    @api.model
+    def create(self, vals):
+        if self.env.context.get('supplier'):
+            vals['name'] = self.env['ir.sequence'].get('rma.order.supplier')
+        else:
+            vals['name'] = self.env['ir.sequence'].get('rma.order')
```

Comment:
IMO, the best way to assign sequence are according these flow:

1. Give a default value for name (e.g. /)
2. Remove readonly from field **name**, so user can override the sequence when they see fit
3. If no change needed (field **name** value == /) then assign sequence when data created

Just an opinion, no need to change if you don't see fit.